### DISCUSSION TITLE
Actions from another repo?

### DISCUSSION THREAD

**Discussion by ChildishGiant** [community/community#25130] [Actions]
<p>I have a unit test I want to go off whenever another repo is pushed/PR submitted. Is there a way to do this?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/childishgiant">@childishgiant</a> ,</p>
<p>You can use ‘<a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#external-events-repository_dispatch" rel="nofollow noopener">repository_dispatch</a>’ event. eg: push/PR submit in repo A, send a repository dispatch event to repo B, to trigger the repoB workflow. </p>
<p>Here is an existing action can help you: <a href="https://github.com/peter-evans/repository-dispatch" rel="nofollow noopener">peter-evans/repository-dispatch@v1</a></p>
<p>Code sample as below:</p>
<p>Repo A:</p>
<pre><code>name: repoAworkflow
on: [push, pull_request]
jobs:
  trigger:
    runs-on: ubuntu-latest
    steps:
      - name: Repository Dispatch
        uses: peter-evans/repository-dispatch@v1
        with:
          token: ${{ secrets.REPO_ACCESS_TOKEN }}
          repository: username/repoB #specify repoB
          event-type: 'repoAevent' #specify a event type
</code></pre>
<p>Repo B:</p>
<pre><code>name: repoBworkflow

on:
  repository_dispatch:
    types: [repoAevent]
...
</code></pre>
